Tasks/AzureWebPowerShellDeployment/Publish-AzureWebDeployment.ps1

```diff
@@ -95,14 +93,67 @@ if($WebSiteLocation)
 $azureCommand = "Publish-AzureWebsiteProject"
 if ($Slot)
 {
-    $azureCommandArguments = "-Name `"$WebSiteName`" -Package `"$packageFile`" -Slot `"$Slot`" $AdditionalArguments"
+    $azureCommandArguments = "-Name `"$WebSiteName`" -Package `"$packageFile`" -Slot `"$Slot`" $AdditionalArguments -ErrorVariable publishAzureWebsiteError"
 }
 else
 {
-    $azureCommandArguments = "-Name `"$WebSiteName`" -Package `"$packageFile`" $AdditionalArguments"
+    $azureCommandArguments = "-Name `"$WebSiteName`" -Package `"$packageFile`" $AdditionalArguments -ErrorVariable publishAzureWebSiteError"
 }
 $finalCommand = "$azureCommand $azureCommandArguments"
 Write-Host "$finalCommand"
 Invoke-Expression -Command $finalCommand
 
+#Update Deployment status - https://github.com/projectkudu/kudu/wiki/REST-API#deployment
+
+if($azureWebSite) {
+    
+    $status = 3 #failed
+    if(!$publishAzureWebsiteError) {
+        $status = 4 #succeeded
+    }
+
+    $username = $azureWebSite.PublishingUsername
+    $securePwd = ConvertTo-SecureString $azureWebSite.PublishingPassword -AsPlainText -Force
+    $credential = New-Object System.Management.Automation.PSCredential ($username, $securePwd)
+    
+    $author = Get-TaskVariable $distributedTaskContext "build.sourceVersionAuthor"
+    $deploymentId = Get-TaskVariable $distributedTaskContext "build.sourceVersion" #let's use commitId as unique deploymentId
+    $message = Get-TaskVariable $distributedTaskContext "build.sourceVersionMessage"
+    $buildId = Get-TaskVariable $distributedTaskContext "build.buildId"
+    
+    $collectionUrl = "$env:SYSTEM_TEAMFOUNDATIONCOLLECTIONURI".TrimEnd('/')
+    $teamproject = "$env:SYSTEM_TEAMPROJECTID"
+    $buildUrl = [string]::Format("{0}/{1}/_build#buildId={2}&_a=summary",$collectionUrl,$teamproject, $buildId)
+    
+    $body = "{
```

Comment:
To handle json special chars:
ConvertTo-Json (New-Object -TypeName psobject -Property @{
  'status' = 'Some status'
  'message' = 'Some message'
})